src/Logger/TideLogsLogger.php

```diff
@@ -186,4 +196,31 @@ public function getSumoLogicCategory() {
     return $category ?: static::DEFAULT_CATEGORY;
   }
 
+  /**
+   * Retrieves all HTTP headers from the current request.
+   *
+   * This function manually extracts HTTP headers from the $_SERVER superglobal.
+   * It processes server variables that begin with 'HTTP_' and converts them
+   * into a format similar to what would be returned by getallheaders(), making it
+   * compatible with environments.
+   *
+   * The function also standardizes header names by replacing underscores with
+   * hyphens and ensuring proper capitalization.
+   *
+   * @return array
+   *   An associative array of HTTP headers where the key is the header name
+   *   (e.g., 'User-Agent') and the value is the header's content.
+   */
+  protected function getAllHeaders() {
+    $headers = [];
+    foreach ($_SERVER as $name => $value) {
+      // Check if the server variable is a header (starts with 'HTTP_').
+      if (substr($name, 0, 5) == 'HTTP_') {
+        // Convert 'HTTP_X_REQUEST_ID' to 'X-Request-Id' format.
+        $headers[str_replace(' ', '-', ucwords(strtolower(str_replace('_', ' ', substr($name, 5)))))] = $value;
```

Comment:
It might make more sense to standardize header names as all lowercase?

Reply:
Done